src/main/java/pw/react/backend/controller/FlatController.kt

```diff
@@ -82,6 +92,102 @@ class FlatController(
         ResponseEntity.badRequest().body(e.message)
     }
 
+
+    @Operation(summary = "Create new flat")
+    @ApiResponse(
+    responseCode = "201",
+    description = "Succesfully created a flat",
+    content = [
+    Content(mediaType = "application/json", schema = Schema(oneOf = [FlatDetailsDto::class]))
+    ]
+    )
+    @ApiResponse(
+    responseCode = "401",
+    description = "Something went wrong"
+    )
+    @PostMapping("/admin/flats")
+    fun postFlat(@RequestBody newFlatDto: NewFlatDto): ResponseEntity<*> =
+    try {
+        val email = newFlatDto.flatowneremail
+        val flatOwnerEntity = flatOwnerRepository.findByEmail(email)
+        var flatOwner: FlatOwner
+        if (flatOwnerEntity == null) {
+            flatOwner = FlatOwner(
+                name = newFlatDto.flatownername,
+                lastName = newFlatDto.flatownerlastName,
+                email = email,
+                phoneNumber = newFlatDto.flatownerphoneNumber,
+                registeredAt = timeProvider().toLocalDateTime(TimeZone.currentSystemDefault()).date,
+            )
+            //val newFlatOwner =  flatOwnerService.save(flatOwner)
+            //flatOwner = newFlatOwner
+        }
+        else{
+            flatOwner = flatOwnerEntity.toDomain()
+        }
+
+
+        val address = Address(
+            street = newFlatDto.street,
+            postalCode = newFlatDto.postalCode,
+            city = newFlatDto.city,
+            country = newFlatDto.country,
+            latitude = newFlatDto.latitude,
+            longitude = newFlatDto.longitude
+        )
+        //val newAddress = addressService.save(address)
+
+        val newFlat = newFlatDto.toDomain(address,flatOwner)
+        val savedFlat = flatService.saveNewFlat(newFlat)
+
+        val price = flatPriceService.savePriceByFlat(savedFlat, newFlatDto.pricePerNight)
+
+        //save facils
+
+        ResponseEntity.ok(savedFlat.id)
+
+    } catch (ex: Exception) {
+    throw FlatValidationException(ex.message, UserController.USERS_PATH)
+    }
+
+    @Operation(summary = "Add an image")
+    @ApiResponse(
+        responseCode = "201",
+        description = "Succesfully added an image",
+        content = [
+            Content(mediaType = "application/json", schema = Schema(oneOf = [FlatDetailsDto::class]))
+        ]
+    )
+    @ApiResponse(
+        responseCode = "401",
+        description = "Something went wrong"
+    )
+    @PostMapping("/admin/flats/{flatId}/image")
+    fun postImage(@PathVariable flatId: String,
+                  @RequestParam("file") file: MultipartFile
+    ): ResponseEntity<*> = try {
+
+        val image = flatImageService.saveImage(flatId, file)
+
+        val fileDownloadUri = ServletUriComponentsBuilder.fromCurrentContextPath()
```

Comment:
`FlatImageService` ma w sobie metodę to generowania urla obrazka - trzeba to tylko tu wywołać.